### PR TITLE
Altering HMM code to run through list of gffs

### PR DESCRIPTION
**Pull request by jdaeth274**
This encompasses some changes I made to the bash script too, this is just to allow it run non-interactively, the RF script has also been tidied to run through in the pipeline and more quietly.

### CODE REVIEW

File: python/pen_checker_cdc.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ def get_options():
     parser.add_argument('--aln', required=True, help='Gene alignment  (required)', type = str)
     parser.add_argument('--tlength', required=True, help='Target length of TPD domain  (required)', type = int)
     parser.add_argument('--tolerance', help='Deviation from expected length allowed', type = int, default = 10)
+    parser.add_argument('--data_dir', help='Location of the data directory of package', type = str, default="./data")
```
Comment (jdaeth274) [inline python/pen_checker_cdc.py:59]: Adding in option for data_directory to allow for running outside the main repo directory

File: python/pen_checker_cdc.py [hunk 2]
```diff
@@ -91,18 +92,24 @@ def get_aln_pos_from_ref(hmm_aln,pos):
...
-        # process file name
-        gff_base = gff.replace('.gff','')
-        
+
+
+        # process file name expecting no extra .
+        gff_base = os.path.basename(fasta)
+        gff_base = re.sub("\..*[a-z,A-Z,0-9].*$","",gff_base)
+        aa_base_name = aa_dir_name + "/" + gff_base
+
```
Comment (jdaeth274) [inline python/pen_checker_cdc.py:112]: Tidying up the output name a bit more for readability and to allow for aa sequences to be dropped in new directory where the script is run from

File: python/pen_checker_cdc.py [hunk 3]
```diff
@@ -152,10 +166,18 @@ def hmm_search_for_gene(fasta,gene):
...
             print(gff_base + '\tTrimethoprim sensitive')
+            status = "S"
         else:
             print(gff_base + '\tTrimethoprim unknown')
+            status = "NA"
+
+    print("Took this long for ORF finder: %s" % (toc_aa_creator - tic_aa_creator))
+    print("Took this long for HMM run: %s" % (toc_hmm_run - tic_hmm_run))
+
+    return status, gff_base
```
Comment (jdaeth274) [inline python/pen_checker_cdc.py:180]: Making sure the search returns the outcome of the HMMER match for resistance, along with the readable isolate id.

File: python/pen_checker_cdc.py [hunk 4]
```diff
@@ -242,7 +264,12 @@ def process_pbp(isolate,protein_fasta,tpd_start,tpd_end,tpd_lab,results_csv,k):
     input_gene_names.append(gene_name.lower())
     all_gene_names = set(input_gene_names)
     gene_length = files_for_input.tlength
-    
+
+    if files_for_input.pbp not in ['pbp1a','pbp2b','pbp2x']:
+        aa_dir_name = "./" + re.sub(".csv","",files_for_input.output) + "_aa_dir"
+        os.mkdir(aa_dir_name)
+        all_gene_names = [files_for_input.pbp]
```
Comment (jdaeth274) [inline python/pen_checker_cdc.py:271]: Making new dir for aa sequences, have to remember to delete for every new run if it doesn't run all the way through

File: python/pen_checker_cdc.py [hunk 5]
```diff
@@ -278,11 +306,18 @@ def process_pbp(isolate,protein_fasta,tpd_start,tpd_end,tpd_lab,results_csv,k):
...
                                                                 gene_rower)
                 else:
-                    hmm_search_for_gene(fasta_file, files_for_input.pbp)
-                    exit(0)
+
+                    current_res, current_isolate = hmm_search_for_gene(fasta_file, files_for_input.pbp, aa_dir_name, files_for_input.data_dir)
+                    gene_id.append(current_res)
+                    isolate_name.append(current_isolate)
+                    skip = True
+                    continue
```
Comment (jdaeth274) [inline python/pen_checker_cdc.py:314]: Getting it to skip through remaining pbp workflow if co-trimoxazole resistance and return resistance and isolate name

File: python/pen_checker_cdc.py [hunk 6]
```diff
@@ -278,11 +306,18 @@ def process_pbp(isolate,protein_fasta,tpd_start,tpd_end,tpd_lab,results_csv,k):
...
                 if correct_length:
                     sys.stderr.write('Found gene ' + gene + ' in ' + bassio_nameo + '\n')
 
+        if skip:
+            continue
+
```
Comment (jdaeth274) [inline python/pen_checker_cdc.py:320]: Skipping rest of loop if co-trimoxazole

**Comment by nickjcroucher** [comment]
Great, thanks!

**Pull request merged**